lib/src/main/java/io/github/davidallison/android/sdk/penremote/SPenRemote.kt

```diff
@@ -76,134 +83,159 @@ object SPenRemote {
 
-        TODO("Validate that air commands + S Pen Framework exist as a package")
+        if (!(Build.BRAND.equals("Samsung", true) &&
+                    Build.MANUFACTURER.equals("Samsung", true))
+        ) {
+            unsupported()
+            return
+        }
+
+//        TODO("Validate that air commands + S Pen Framework exist as a package")
         // use INTENT_PACKAGE_NAME and FLAG_GRANT_PREFIX_URI_PERMISSION
         // If an exception is not returned, the check succeeded
         // If a NameNotFoundException is obtained, log, send UNSUPPORTED_DEVICE and return
 
-        TODO("Validate that Bluetooth Low Energy is supported [SemFeatures.HAS_BLUETOOTH_LOW_ENERGY]")
-        // If not, log, send UNSUPPORTED_DEVICE and return
-
-
-        TODO("Verify supported features; See: isFeatureEnabled")
-        // load supported features and ensure the call works
-        // If not, log, send UNSUPPORTED_DEVICE and return
-
-        TODO("We can start binding")
-
-        TODO("The provided [listener] should be usable inside [ServiceConnection]")
-        TODO("Log that we're binding")
-
+        try {
+            activity.packageManager.getApplicationInfo(
+                Constants.INTENT_PACKAGE_NAME,
+                PackageManager.GET_META_DATA
+            )
+        } catch (e: PackageManager.NameNotFoundException) {
+            unsupported()
+            return
+        }
 
-        TODO("An intent is produced, for a call to bindService")
-        // className: INTENT_PACKAGE_NAME, INTENT_CLASS_NAME
-        // extras:
-          // INTENT_EXTRA_BINDER_TYPE           // constant value
-          // INTENT_EXTRA_CLIENT_VERSION        // constant value
-          // INTENT_EXTRA_CLIENT_PACKAGE_NAME   // dynamic value obtained from the provided context
 
-        TODO("bindService is called, using the above intent and flag: BIND_AUTO_CREATE")
-        // If a SecurityException is caught from the `bindService` call
-        //    Log an error stating that the following permission is required: "com.samsung.android.sdk.penremote.BIND_SPEN_REMOTE"
-        //    BUG: if this error occurs neither ConnectionResultCallback nor
-        //    ConnectionStateChangeListener are called
+        if (!FloatingFeatureReflected().getBoolean(SemFeatures.HAS_BLUETOOTH_LOW_ENERGY)) {
+            unsupported()
+            return
+        }
 
-        // if the call succeeds, [isConnected] should return true
+        if (!isFeatureEnabled(Feature.FEATURE_TYPE_BUTTON)) {
+            unsupported()
+            return
+        }
 
-        // execution continues asynchronously inside the [ServiceConnection] instance which is
-        // managed by the class and passed to `bindService`
+        connectionResultCallback = listener
+
+        @SuppressLint("WrongConstant")
+        val intent = Intent()
+            .addFlags(Constants.INTENT_FLAGS)
+            .setClassName(Constants.INTENT_PACKAGE_NAME, Constants.INTENT_CLASS_NAME)
```

Comment:
`Constants` only existed for documentation purposes, feel free to move them to variables, or inline them as you see fit